pennylane/templates/broadcast.py

```diff
@@ -13,49 +13,59 @@
 # limitations under the License.
 r"""
 Contains the ``broadcast`` template constructor.
+
+To add a new structure, extend the variables ``OPTIONS``, ``n_parameters`` and ``wire_sequence``, and
+update the list of parameter numbers in the docstring. Optionally add a usage example at the end of the
+``UsageDetails`` directive.
 """
 # pylint: disable-msg=too-many-branches,too-many-arguments,protected-access
 from collections import Iterable
 
 from pennylane.templates.decorator import template
-from pennylane.templates.utils import _check_wires, _check_type, _get_shape
+from pennylane.templates.utils import _check_wires, _check_type, _get_shape, _check_is_in_options
 
 
 @template
-def Broadcast(block, wires, parameters=None, kwargs={}):
-    r"""Applies a (potentially parametrized) single-qubit unitary ``block`` to each wire.
+def broadcast(block, wires, structure="single", parameters=None, kwargs={}):
+    r"""Applies a unitary multiple times to a specific pattern of wires.
+
+    The unitary ``block`` is either a quantum operation (such as :meth:`~.pennylane.ops.RX`), or a
+    user-supplied template. Depending on the chosen structure, ``block`` is applied to a wire or a subset of wires:
 
-    .. figure:: ../../_static/templates/constructors/broadcast.png
-        :align: center
-        :width: 20%
-        :target: javascript:void(0);
+    * ``structure= 'single'`` (the default case) applies a single-wire block to each one of the :math:`M` wires:
 
-    If the block does not depend on parameters, ``parameters`` is set to ``None``.
+      .. figure:: ../_static/templates/broadcast_single.png
```

Comment:
The reason not to do this previously was that it is actually *not* the same unitary, because it gets different parameters each time it is applied. So the cleaner way would be to use :math:`U(\theta^1_1, \dots \theta^1_D)`, :math:`U(\theta^2_1, \dots \theta^2_D)` ,..., and then things get very messy.

What do you think?

Reply:
That's a fair point, yes! One way would be to simply denote it as `theta` and regard it as a symbolic variable. But then it might actually confuse people into thinking that each time the same arguments are passed to the gate. :thinking: 

Perhaps, after all, it is just best to leave it as is, as you suggested. :)